core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -24,27 +24,32 @@ import com.fasterxml.jackson.databind.annotation.JsonDeserialize
 
 import org.apache.spark.JobExecutionStatus
 
-class ApplicationInfo private[spark](
-    val id: String,
-    val name: String,
-    val coresGranted: Option[Int],
-    val maxCores: Option[Int],
-    val coresPerExecutor: Option[Int],
-    val memoryPerExecutorMB: Option[Int],
-    val attempts: Seq[ApplicationAttemptInfo])
+case class ApplicationInfo private[spark](
```

Comment:
What is turning this into a case class achieving? I only see noise caused by this change.

Reply:
We check equality of ApplicationInfos during testing. I think converting them to case classes introduce less complexity than adding a proper equals method (ditto for ApplicationAttemptInfo below). 
In addition we copy ApplicationInfos in FsHistoryProvider:819, it's also simpler with case classes.